### PR TITLE
Stealer rules targeting file-manager software

### PR DESCRIPTION
**Pull request by re-fox**
Discussed a bit in https://github.com/fireeye/capa-rules/issues/336 .  Thanks for the input @williballenthin 

> I recently worked my way through an old stealer/grabber (5a2f620f29ca2f44fc22df67b674198f) with the intent to build out capa-rules.
>
>While doing this exercise, I found that many of the rules are subjectively "low quality" relying on pairings of strings.
>
>I wanted to share these rules, in case they may be useful. I didn't roll these up into a giant PR until they received some peer review or qualitative analysis. Capa's scope (in this case, all function level) results in the rules not being too prone to FP's,
>
>Let me know if these rules would be a welcome addition to the capa-rules main repository, and if not, I'll keep a fork with supplemental rules. If a PR is decided, the rules have been linted with --thorough and capafmt has been applied.

I'll upload `5a2f620f29ca2f44fc22df67b674198f` in capa-testfiles  (https://github.com/fireeye/capa-testfiles/pull/85)

While some of these rules may be sparse, they could always serve as a placeholder if another (more robust) sample becomes available.  Some of the more obscure software doesn't have `references`, either because the company is defunct or the software is no longer available.

### CODE REVIEW

File: collection/file-managers/gather-coreftp-information.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    examples:
+      - 5a2f620f29ca2f44fc22df67b674198f:0x4063FD
+  features:
+    - or:
+      - string: /Software\\FTPWare\\COREFTP\\Sites/
+      - and:
+        - string: "Host"
+        - string: "User"
+        - string: "Port"
+        - string: "PthR"
```
Comment (williballenthin) [inline collection/file-managers/gather-coreftp-information.yml:20]: if not for this last string, the `and` here might be loose. looks ok though.

File: collection/file-managers/gather-direct-ftp-information.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  features:
+    - or:
+      - string: /Software\\CoffeeCup Software\\Internet\\Profiles/
+      - string: /\\CoffeeCup Software/
+      - and:
+        - string: "Password"
+        - string: "HostName"
+        - string: "Port"
+        - string: "Username"
+        - string: "HostDirName"
```
Comment (williballenthin) [inline collection/file-managers/gather-direct-ftp-information.yml:22]: *maybe* loose? *shrug*

File: collection/file-managers/gather-filezilla-information.yml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    - or:
+      - and:
+        - string: /\\sitemanager\.xml/
+        - string: /\\recentservers\.xml/
+        - string: /\\filezilla.xml/
+      - and:
+        - string: /Software\\FileZilla/
+        - string: "Install_Dir"
+        - string: /Software\\FileZilla Client/
+      - 3 or more:
```
Comment (williballenthin) [inline collection/file-managers/gather-filezilla-information.yml:23]: cool

File: collection/file-managers/gather-xftp-information.yml [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+    att&ck:
+      - Credential Access::Credentials from Password Stores [T1555]
+    references:
+      - https://www.netsarang.com/en/xftp-download/
+    examples:
+      - 5a2f620f29ca2f44fc22df67b674198f:0x40CBEE
+  features:
+    - and:
+      - string: ".xfp"
+      - string: /\\NetSarang/
```
Comment (williballenthin) [inline collection/file-managers/gather-xftp-information.yml:16]: :-)

**Review by williballenthin — APPROVED** [review]
wow, 48 rules!?

nice job.

i have no major concerns here. if we happen to eventually encounter an FP, then we can be aggressive about disabling the rule. as is, it seems unlikely.

thanks!

**Comment by williballenthin** [comment]
lets get one other review just to ensure awareness around the team. should be merged by monday.

**Comment by re-fox** [comment]
Thanks for looking through these.  

I did run these rules against the capa-testfiles repo and they only trigger on the sample that is referenced.   From a false-positive perspective that is a good sign.  I have the rules running against a larger corpus and will be keeping a watch for any noise, but I don't expect much.

File: collection/file-managers/gather-alftp-information.yml [hunk 5]
```diff
@@ -0,0 +1,17 @@
+rule:
+  meta:
+    name: gather alftp information
+    namespace: collection/file-managers
+    author: "@_re_fox"
+    scope: function
+    att&ck:
+      - Credential Access::Credentials from Password Stores [T1555]
+    references:
+      - https://www.altools.co.kr/Main/Default.aspx
```
Comment (Ana06) [inline collection/file-managers/gather-alftp-information.yml:10]: This page is not in English. Is there a link to an English version?

File: collection/file-managers/gather-cuteftp-information.yml [hunk 6]
```diff
@@ -0,0 +1,19 @@
+rule:
+  meta:
+    name: gather cuteftp information
+    namespace: collection/file-managers
+    author: "@_re_fox"
+    scope: function
+    att&ck:
+      - Credential Access::Credentials from Password Stores [T1555]
+    references:
+      - https://www.globalscape.com/cuteftp
```
Comment (Ana06) [inline collection/file-managers/gather-cuteftp-information.yml:10]: This page seems to be down
> The website encountered an unexpected error. Please try again later.
Error: Class 'Drupal\field_collection\Plugin\Field\FieldType\FieldCollection' not found in Drupal\field\Entity\FieldStorageConfig->getSchema() (line 453 of core/modules/field/src/Entity/FieldStorageConfig.php).

File: collection/file-managers/gather-ftpinfo-information.yml [hunk 7]
```diff
@@ -0,0 +1,19 @@
+rule:
+  meta:
+    name: gather ftpinfo information
+    namespace: collection/file-managers
+    author: "@_re_fox"
+    scope: function
+    att&ck:
+      - Credential Access::Credentials from Password Stores [T1555]
+    references:
+      - https://www.ftpinfo.ru/
```
Comment (Ana06) [inline collection/file-managers/gather-ftpinfo-information.yml:10]: This page is not in English. Is there a link to an English version?

**Review by Ana06 — APPROVED** [review]
48 rules!!! 🎉 Awesome!

File: collection/file-managers/gather-cuteftp-information.yml (hunk 6 above)
Comment (re-fox) [reply collection/file-managers/gather-cuteftp-information.yml:11]: I think that may be a recent error on the company page.  It's listed as the primary link.  I added an additional Wikipedia link in https://github.com/fireeye/capa-rules/pull/337/commits/a3899f67dce9d41cfe736c8e993e10caa854756f

File: collection/file-managers/gather-ftpinfo-information.yml (hunk 7 above)
Comment (re-fox) [reply collection/file-managers/gather-ftpinfo-information.yml:10]: I searched and have not found any English description of the software.

**Comment by williballenthin** [comment]
thank you!

**Comment by mr-tz** [comment]
Thank you!

**Pull request merged**